test/unit-test/mqtt_agent_utest.c

```diff
@@ -30,25 +30,80 @@
 #include "unity.h"
 
 /* Include paths for public enums, structures, and macros. */
-
+#include "mqtt_agent.h"
 #include "mock_core_mqtt.h"
+#include "mock_core_mqtt_state.h"
 
 /**
- * @brief MQTT client identifier.
+ * @brief Time at the beginning of each test. Note that this is not updated with
+ * a real clock. Instead, we simply increment this variable.
  */
-#define MQTT_CLIENT_IDENTIFIER                 "testclient"
+static uint32_t globalEntryTime = 0;
+
+/* ========================================================================== */
 
 /**
- * @brief A sample network context that we set to NULL.
+ * @brief A mocked timer query function that increments on every call.
  */
-#define MQTT_SAMPLE_NETWORK_CONTEXT            ( NULL )
+static uint32_t getTime( void )
+{
+    return globalEntryTime++;
+}
+/* ========================================================================== */
 
 /**
- * @brief Time at the beginning of each test. Note that this is not updated with
- * a real clock. Instead, we simply increment this variable.
+ * @brief A mocked send function to send command to agent.
  */
-static uint32_t globalEntryTime = 0;
+static bool( send )( AgentMessageContext_t * pMsgCtx,
+                     const void * pData,
+                     uint32_t blockTimeMs )
+{
+    return true;
+}
 
+/**
+ * @brief A mocked receive function for the agent to receive command.
+ */
+static bool( receive )( AgentMessageContext_t * pMsgCtx,
+                        void * pBuffer,
+                        uint32_t blockTimeMs )
+{
+    return true;
+}
+
+/**
+ * @brief A mocked function to obtain a pointer to an allocated command.
+ */
+static bool( commandRelease )( Command_t * pCommandToRelease )
+{
+    return true;
+}
+
+/**
+ * @brief A mocked function to release an allocated command.
+ */
+static Command_t * ( getCommand ) ( uint32_t blockTimeMs )
+{
+    Command_t * command;
+    return command;
+}
+
+/**
+ * @brief A mocked Callback to invoke upon command completion.
+ */
+static void( CommandCallback )( void * pCmdCallbackContext,
+                                MQTTAgentReturnInfo_t * pReturnInfo )
+{
+    return;
+}
+
+/**
+ * @brief Context Definition with which tasks may deliver messages to the agent.
+ */
+struct AgentMessageContext
+{
+    int mock;
```

Comment:
I'd suggest changing this to something that would be more useful when testing `MQTTAgent_CommandLoop`
```suggestion
    Command_t * pSentCommand;
```
Then `send` can set this pointer and `receive` can read from it